simulate-event.js

```diff
@@ -263,6 +272,25 @@ module.exports = function (element, type, options) {
   }
 
   var eventType = eventTypes[type];
+
+  // Handle parameters which must be manually overridden using
+  // Object.defineProperty.
+  var overrides = {};
+  if (eventType === 'KeyboardEvent' && options) {
+    overrides['keyCode'] = options['keyCode'] || 0;
+    overrides['key'] = options['key'] || '';
+    overrides['which'] = options['which'] || overrides['keyCode'];
+  }
```

Comment:
This is better. Can it be moved to a function instead though? I'll write an example:

```js
function overrides (eventType, options) {
  if (eventType === 'KeyboardEvent' && options) {
    return { keyCode: options.keyCode || 0, ... }
  }
}

var eventOptions = extend(
  {
    bubbles:    true,
    cancelable: true
  }, 
  result(eventOptions, eventType, element, type, options), 
  options, 
  overrides(eventType, options)
)
```